### PR TITLE
[Cleanup] Drop VPN Diffie–Hellman parameters

### LINKED ISSUE #9093: [Cleanup] VPN Diffie–Hellman parameters no longer needed
**How to categorize this issue?**

/area networking security
/kind cleanup

**What would you like to be added**:
Vpn release v0.23.0 that contains https://github.com/gardener/vpn2/pull/53 was published and merged into g/g. That release removes the need for supplying the vpn server with dh parameters. This means that we can remove https://github.com/gardener/gardener/blob/6d43f8c4754c11c4a9619ee8386b2878db682f37/pkg/operation/botanist/vpnseedserver.go#L31 and other related code from this repository. For the sake of backwards compatibility the code can be left for another release or two in case there are new Gardener installations that use old VPN release.

### PR DESCRIPTION
**Pull request by ary1992**
**How to categorize this PR?**

/area quality
/kind cleanup

**What this PR does / why we need it**:
Drop no longer needed VPN Diffie–Hellman parameters. [Vpn](https://github.com/gardener/vpn2) release `v0.23.0` removed the need for supplying the vpn server with `Diffie–Hellman` parameters. Hence the related codes are dropped from our code-base.

`vpn-seed-server-dh`  secret will be cleaned up the GRM garbage-collector controller as this secret has `resources.gardener.cloud/garbage-collectable-reference: "true"` label set.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener/issues/9093

**Special notes for your reviewer**:

**Release note**:

```other operator
Secret `vpn-seed-server-dh` will no longer be created as [gardener/vpn2](https://github.com/gardener/vpn2) release `v0.23.0` removed the need for supplying the vpn server with `Diffie–Hellman` parameters.
```

```noteworthy operator
Secret `openvpn-diffie-hellman-key` in the `garden` namespace containing the Diffie-Hellmann key can be deleted from landscapes as it is no longer needed.
```

### CODE REVIEW

**Review by ScheererJ — COMMENTED** [review]
Thanks for cleaning up.

/lgtm

**Comment by ScheererJ** [comment]
Could you please check the tests and run `make generate`?

**Review by rfranzke — CHANGES_REQUESTED** [review]
Can you delete https://github.com/gardener/gardener/blob/master/example/10-secret-openvpn-diffie-hellman.yaml and adapt https://github.com/gardener/gardener/blob/932601d6cde467b05baede8dc7aff2f46d63dc26/docs/operations/configuration.md?plain=1#L75?

**Comment by rfranzke** [comment]
Also, please adapt the release note for operators (i.e., they can delete this secret from garden namespace, etc.)

File: pkg/component/networking/vpn/seedserver/seedserver.go [hunk 1]
```diff
@@ -195,19 +182,9 @@ func (v *vpnSeedServer) Deploy(ctx context.Context) error {
 				fileNameEnvoyConfig: v.getEnvoyConfig(),
 			},
 		}
-
-		dhSecret = &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      secretNameDH,
```
Comment (ialidzhikov) [inline pkg/component/networking/vpn/seedserver/seedserver.go:201]: Nit: You could also drop the const in https://github.com/gardener/gardener/blob/be23065e1441a703eef7f2ed565c865ab9f43017/pkg/component/networking/vpn/seedserver/seedserver.go#L76.

File: pkg/component/networking/vpn/seedserver/seedserver.go [hunk 2]
```diff
@@ -386,10 +359,6 @@ func (v *vpnSeedServer) podTemplate(configMap *corev1.ConfigMap, dhSecret, secre
 							Name:      volumeNameTLSAuth,
 							MountPath: volumeMountPathTLSAuth,
 						},
-						{
-							Name:      volumeNameDH,
-							MountPath: volumeMountPathDH,
```
Comment (ialidzhikov) [inline pkg/component/networking/vpn/seedserver/seedserver.go:391]: Nit: You could also drop https://github.com/gardener/gardener/blob/be23065e1441a703eef7f2ed565c865ab9f43017/pkg/component/networking/vpn/seedserver/seedserver.go#L92 and https://github.com/gardener/gardener/blob/be23065e1441a703eef7f2ed565c865ab9f43017/pkg/component/networking/vpn/seedserver/seedserver.go#L85.

**Review by ialidzhikov — COMMENTED** [review]
We had a look with @dimitar-kostadinov and @Kostov6.
 Minor cleanup suggestions, otherwise lgtm.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**